src/main/java/org/cactoos/iterable/IterableOfInts.java

```diff
@@ -40,13 +40,8 @@
      * @param values Integer values
      */
     public IterableOfInts(final int... values) {
-        super(() -> {
-            final Collection<Integer> iterable =
-                new ArrayList<>(values.length);
-            for (final int value: values) {
-                iterable.add(value);
-            }
-            return iterable;
-        });
+        super(() -> ()
+            -> new UncheckedScalar<>(() -> new IteratorOfInts(values)).value()
```

Comment:
@krzyk I might be wrong here (please, correct me if it's so), but using of `UncheckedScalar` looks redundant here, as the `super` ctor already wraps the argument into `UncheckedScalar`: https://github.com/yegor256/cactoos/blob/5ead4ac1184cfb35f848a8e8f1f181ac70a9270f/src/main/java/org/cactoos/iterable/IterableEnvelope.java#L54
Could you check please if that's correct and UncheckedScalar can be ommitted here?

Reply:
@proshin-roman unchecked scalar is for `Iterator` interface (there are 3 lambdas here) so I need to wrap it at the bottom, wrapping in IterableEnvelope doesn't help here

Reply:
@krzyk why didn't you just write `() -> () -> new IteratorOfInts(values)`? I don't get what you wrote provides. The constructor of `IteratorOfInts` does not throw any exception...

Reply:
@krzyk or `() -> new IterableOf<>(new IteratorOfInts(values))` if you prefer

Reply:
@victornoel I didn't know I could (regarding the first of your comments - it wasn't possible, I didn't tried the second one)

Reply:
@krzyk ok thanks, I will take that into account for solving #838 :)